### PR TITLE
Fixed #47 and Worked on #48 -

### LINKED ISSUE #47: Use Redux Actions
Instead of creating a js object for actions, use redux actions.

### PR DESCRIPTION
**Pull request by parikhv15**
* Added redux-actions
* Refactored Test cases for using redux-actions
* Added testcase for moving, but needs refactoring
* Added action text to constants

### CODE REVIEW

File: lcov.info [hunk 1]
```diff
@@ -55,97 +73,110 @@ DA:0,4
...
+BRDA:19,4,1,0
+BRDA:19,4,2,4
+BRDA:19,4,3,0
+BRDA:35,5,0,2
+BRDA:35,5,1,2
+BRDA:70,6,0,0
+BRDA:70,6,1,0
+BRF:14
+BRH:10
 end_of_record
```
Comment (jwbennet) [inline lcov.info]: This file should have been .gitignore'd

File: src/actions/index.js [hunk 2]
```diff
@@ -1,42 +1,43 @@
...
+  },
+  getMoveTodo: (sourceTodo, targetTodo) => {
+    return {
+      id: sourceTodo.id,
+      text: sourceTodo.text,
+      completed: sourceTodo.completed,
+      target_id: targetTodo.id
+    }
+  },
+  getVisibilityFilter: (filter) => {
```
Comment (jwbennet) [inline src/actions/index.js]: You shouldn't need to have this payload creator specified like this.  The default payload creator will take in an argument and put it in the payload.

File: src/components/Link.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const Link = ({ active, children, onClick }) => {
   return (
     <a href="#"
        onClick={e => {
-         e.preventDefault()
+         e.preventDefault();
```
Comment (jwbennet) [inline src/components/Link.js]: Are issues like this one being verified with your linter?

File: src/containers/VisibleTodoList.js [hunk 4]
```diff
@@ -10,13 +10,14 @@ const getVisibleTodos = (todos, filter) => {
...
-      return todos.filter(t => t.completed);
+      return todos.filter(t => t.payload.completed);
     case 'SHOW_ACTIVE':
-      return todos.filter(t => !t.completed);
+      return todos.filter(t => !t.payload.completed);
   }
 };
 
 const mapStateToProps = (state) => {
+  console.log(state);
```
Comment (jwbennet) [inline src/containers/VisibleTodoList.js]: Should not commit console.log statements

File: src/reducers/todos.js [hunk 5]
```diff
@@ -1,24 +1,26 @@
...
  */
 const todo = (state, action) => {
   switch (action.type) {
-    case 'ADD_TODO':
+    case Actions.addTodo:
       return {
-        id: action.id,
-        text: action.text,
+        id: action.payload.id,
+        text: action.payload.text,
```
Comment (jwbennet) [inline src/reducers/todos.js]: You could destructure the action here to make this a bit more streamlined.

File: test/reducers/reducer-test.js [hunk 6]
```diff
@@ -107,5 +115,71 @@ describe('todos reducer', () => {
...
+  //      completed: false,
+  //      id: 1
+  //    },
+  //    {
+  //      text: 'Run the tests',
+  //      completed: false,
+  //      id: 0
+  //    }
+  //  ])
+  //})
```
Comment (jwbennet) [inline test/reducers/reducer-test.js]: Remove commented out code

File: src/actions/index.js (hunk 2 above)
Comment (parikhv15) [inline src/actions/index.js]: Yes, but i am modifying default arguments so as to create a specific object, like in the case of moveTodo. Also, each todo needs an unique id. So should i just have default arguments in the payload instead?

File: src/containers/VisibleTodoList.js (hunk 4 above)
Comment (parikhv15) [inline src/containers/VisibleTodoList.js]: Yes i will. Sorry about that. Missed removing it before the push.

File: test/reducers/reducer-test.js (hunk 6 above)
Comment (parikhv15) [inline test/reducers/reducer-test.js]: Yes i will. This test case was not being passed and was throwing some error. So i had commented it out. Sorry, should have removed it.

File: lcov.info (hunk 1 above)
Comment (parikhv15) [inline lcov.info]: i did add lcov.info to .gitignore, but some how it didnt get sync. Will look into it and remove it from here.

File: src/reducers/todos.js (hunk 5 above)
Comment (parikhv15) [inline src/reducers/todos.js]: so should i have different functions for each actions here?

Comment (jwbennet) [inline src/reducers/todos.js]: Not necessarily, I was thinking something like this: ```
const todo = (state, { type, payload: { id, text, completed } }) {
  switch(type) {
    return {
      id,
      text,
      completed: false
    }
    ...
  }
}```

Comment (parikhv15) [inline src/reducers/todos.js]: Okay.Yes that looks better. :)
I will make necessary changes. Thanks James.

**Pull request merged**